codes/classical/bits/bch.yml

```diff
@@ -8,21 +8,28 @@ physical: bits
 logical: bits
 
 name: 'Bose–Chaudhuri–Hocquenghem (BCH) code'
-short_name: 'BCH'
-introduced: '\cite{doi:10.1016/s0019-9958(60)90287-4}'
-
-description: 'Stub.'
-
-relations:
-  parents:
-    - code_id: alternant
-      detail: ''
-
-  cousins:
-    - code_id: reed_solomon
-      detail: ''
-    - code_id: cyclic
-      detail: ''
-    - code_id: goppa
-      detail: ''
-# MacWilliams Sloane Ch. 12
+introduced: '\cite{doi:10.1016/S0019-9958(60)90287-4}'
+
+description: 'Bose–Chaudhuri–Hocquenghem (BCH) codes can be defined as subfield codes of the Reed-Solomon codes. If \( C_{\text{RS}} \) is a Reed-Solomon code containing elements in the finite field \( \text{GF}(2^{m}) \) for some \(m \geq 0 \), then the corresponding BCH code is defined as the code \( C_{\text{BCH}} = C_{\text{RS}} \cap (\text{GF}(2))^n \). That is, the BCH codes contains only those Reed-Solomon codewords whose components are elements of \( \text{GF}(2) \).'
```

Comment:
This is not fully accurate. q-ary BCH codes of length n=q^m -1 can be defined as subfield subcodes of q^m-ary RS codes of length q^m-1. 

A more traditional definition of a BCH code is as follows. A q-ary BCH code of length n=q^m-1 with designed distance d is a cyclic code with zeros alpha, alpha^2,...,alpha^d-1, where alpha is a primitive element of the field q^m-1.

Reply:
I'll use the definition of BCH codes you've mentioned.